src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -455,6 +496,45 @@ bool EngineShoutcast::serverConnect() {
 
 void EngineShoutcast::write(unsigned char *header, unsigned char *body,
                             int headerLen, int bodyLen) {
+    struct shoutcastCacheObject *l_SCacheObj = (struct shoutcastCacheObject *) malloc(sizeof(struct shoutcastCacheObject));
```

Comment:
malloc and free are potentially blocking system calls. We should not use them here regularly.   
Can we preallocate the memory? 
Nit: break long lines <= cloumn 80